pkg/policyprovider/configpolicy/configpolicy_test.go

```diff
@@ -80,3 +86,168 @@ func TestPolicyEnforcer_ContinueVerifyOnFailure(t *testing.T) {
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    false,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: true,
+		},
+		{
+			// all notary artifact policy but only 1 notary report is true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "all",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    false,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: false,
+		},
+		{
+			// all notary artifact policy and both notary reports are true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "all",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: true,
+		},
+		{
+			// any notary artifact policy, any sbom artifact policy and notary report is true and sbom is false
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "any",
+					"org.example.sbom.v0":            "any",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    false,
+					ArtifactType: "org.example.sbom.v0",
+				},
+			},
+			output: false,
+		},
+		{
+			// any notary artifact policy, all sbom artifact policy, and both notary and sbom are true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "any",
+					"org.example.sbom.v0":            "all",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "org.example.sbom.v0",
+				},
+			},
+			output: true,
+		},
```

Comment:
Can we add a test case for cosign signature? how does the configuration work when customer is sharing a configuration between cosign/notary

Reply:
or maybe this is something we need to call out that we don't support?

Reply:
At the policy level, the customer can specify artifact policy for the notary and cosign signature together. This would mean for an overall success, at least one cosign and notary signature would have to be provided. But then the subject image would need to be a public registry since currently if cosign enabled is set to true, it'll return a 401. I definitely agree we need to call this mixed cased and private registry cosign verification in general. I was thinking of adding a warning here: https://github.com/deislabs/ratify/blob/main/plugins/verifier/cosign/README.md